config/checker.go

```diff
@@ -2,45 +2,63 @@ package config
 
 import (
 	"fmt"
-	"github.com/shitpostingio/autopostingbot/config/structs"
 	"reflect"
+
+	"github.com/shitpostingio/autopostingbot/config/structs"
 )
 
 // checkMandatoryFields uses reflection to see if there are
 // mandatory fields with zero value.
+// When using with isReload flag, only checks for fields with reloadable:"true" tag
 func checkMandatoryFields(isReload bool, config structs.Config) error {
-	return checkStruct(isReload, reflect.TypeOf(config), reflect.ValueOf(config))
+	errors := ""
+	for _, err := range checkStruct(isReload, "root", reflect.TypeOf(config), reflect.ValueOf(config)) {
+		errors += fmt.Sprintln(err.Error())
```

Comment:
Instead of using both Sprintln and concatenation we could use `errors = fmt.Sprintf("%s%s\n", errors, err.Error())` to have the same effect